docs/src/modules/ROOT/pages/using-timefold-solver/modeling-planning-problems.adoc

```diff
@@ -981,7 +980,214 @@ capacity: Annotated[int, PiggybackShadowVariable(shadow_variable_name="transport
+|"collectionVar[].varName"
+|Refers to a variable (`varName`) on each element in a collection (`collectionVar`) on the entity. The collection must not change during solving.
+|===
+
+Concrete example
+[tabs]
+====
+Java::
++
+[source,java,options="nowrap"]
+----
+@PlanningEntity
+public class Job {
+
+    @PreviousElementShadowVariable
+    Job previous;
+
+    private int durationInDays;
+
+    private Collection<Job> dependencies;
+
+    @ShadowVariable(supplierName="startDateSupplier")
+    LocalDateTime startDate;
+
+    @ShadowVariable(supplierName="endDateSupplier")
+    private LocalDate endDate;
+
+    @ShadowSources({"previous.endDate", "dependencies[].endDate"})
+    public LocalDate startDateSupplier() {
+      LocalDateTime readyTime = null;
+      if (previous != null) {
+          readyTime = previous.endTime;
+      } else {
+          return null;
+      }
+      if (dependencies != null) {
+          for (var dependency : dependencies) {
+              if (dependency.endTime == null) {
+                  return null;
+              }
+              readyTime = readyTime.isBefore(dependency.endTime) ? dependency.endTime: readyTime;
+          }
+      }
+      return readyTime;
+    }
+
+    @ShadowSources("startDate")
+    public LocalDate endDateSupplier() {
+        if (startDate == null) {
+            return null;
+        } else {
+            return startDate.plusDays(durationInDays);
+        }
+    }
+
+}
+----
+
+Python::
++
+[source,python,options="nowrap"]
+----
+# Currently not supported in Python.
+----
+====
+
+[cols="1,1,2", options="header"]
+|===
+|Reference
+|Form
+|Meaning
+
+|"startDate"
+|Simple Variable Name
+|Direct reference to the `startDate` field.
+
+|"previous.endDate"
+|Chained Property Path
+|Accesses `endDate` from the `previous` variable.
+
+|"dependencies[].endTime"
+|Collection Element Access
+|Accesses `endTime` of each element in the `dependencies` collection.
+|===
+
+
+==== Detecting Loops in Shadow Variables
+
+In certain cases, shadow variables may form an infinite loop.
+When this occurs, the solver detects the cycle, breaks the loop, and assigns `null` to the involved shadow variables.
+
+A supplier variable is considered part of a _loop_ if:
+
+- It directly or indirectly depends on itself.
+For example, if variable `a` depends on `b` and `b` depends on `a`, both are in a loop.
+- It depends on another variable that is already part of a loop.
+For example, if `c` depends on `a`, and `a` is in a loop with `b`, then `c` is also looped.
+
+To detect whether an entity has been affected by a shadow variable loop, annotate a boolean field with `@ShadowVariableLooped`.
+The solver will set this field to true if the entity is part of a loop.
+
+[tabs]
+====
+Java::
++
+[source,java,options="nowrap"]
+----
+@PlanningEntity
+public class Job {
+    @ShadowVariableLooped
+    boolean isLooped;
+}
+----
+
+Python::
++
+[source,python,options="nowrap"]
+----
+# Currently not supported in Python.
```

Comment:
Maybe to save Python people's time, we mention this in the beginning of the section, next to the preview note?

Reply:
Good idea, will add the note as well.